NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/PortTranslation.md

```diff
@@ -0,0 +1,129 @@
+```
+
+Output:
+
+```
+Chain PREROUTING (policy ACCEPT 0 packets, 0 bytes)
+ pkts bytes target     prot opt in     out     source               destination
+    0     0 DNAT       tcp  --  *      *       0.0.0.0/0            91.245.77.1          tcp dpt:80 to:192.168.0.2:8080
+
+Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
+ pkts bytes target     prot opt in     out     source               destination
+
+Chain OUTPUT (policy ACCEPT 0 packets, 0 bytes)
+ pkts bytes target     prot opt in     out     source               destination
+
+Chain POSTROUTING (policy ACCEPT 0 packets, 0 bytes)
+ pkts bytes target     prot opt in     out     source               destination
+    0     0 SNAT       tcp  --  *      *       192.168.0.2          0.0.0.0/0            tcp spt:8080 to:91.245.77.1:80
+```
+
+**Notes**
+
+- **PREROUTING Chain:** Alters the destination IP address and port of incoming packets before they are routed. This is useful for forwarding incoming requests to a different internal server or port.
+- **POSTROUTING Chain:** Alters the source IP address and port of outgoing packets after routing. This ensures that the response packets go back through the NAT device and then to the correct external client.
+- **DNAT (Destination NAT):** Changes the destination address of packets.
+- **SNAT (Source NAT):** Changes the source address of packets.
+
+## Private to Private Flow
+
+To avoid translating packets within the private subnet, add a rule to bypass NAT for such traffic:
+
+```
+# Bypass NAT for private-to-private traffic
+iptables -t nat -A POSTROUTING -s 192.168.0.0/24 -d 192.168.0.0/24 -j ACCEPT
+```
+
+```
+Chain PREROUTING (policy ACCEPT 0 packets, 0 bytes)
+ pkts bytes target     prot opt in     out     source               destination
+    0     0 DNAT       tcp  --  *      *       0.0.0.0/0            91.245.77.1          tcp dpt:80 to:192.168.0.2:8080
+
+Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
+ pkts bytes target     prot opt in     out     source               destination
+
+Chain OUTPUT (policy ACCEPT 0 packets, 0 bytes)
+ pkts bytes target     prot opt in     out     source               destination
+
+Chain POSTROUTING (policy ACCEPT 0 packets, 0 bytes)
+ pkts bytes target     prot opt in     out     source               destination
+    0     0 SNAT       tcp  --  *      *       192.168.0.2          0.0.0.0/0            tcp spt:8080 to:91.245.77.1:80
+    0     0 ACCEPT     all  --  *      *       192.168.0.0/24       192.168.0.0/24
+```
+
+Consider the following setup:
+
+- Private Host (192.168.0.2) wants to be accessible from a Public IP (91.245.77.1) on port 80.
+- The actual service on the Private Host is running on port 8080.
+
+With the above iptables configuration:
+
+- Incoming traffic on 91.245.77.1:80 is forwarded to 192.168.0.2:8080.
+- Outgoing traffic from 192.168.0.2:8080 appears as originating from 91.245.77.1:80.
+
```

Comment:
Yes, you've typed these descriptions but I think it works better if you had a graph of sample topology and show the config works (i.e. add a simple command like ping and show it is successful.)